zerver/lib/email_mirror.py

```diff
@@ -412,20 +411,12 @@ def key_fragment(self) -> str:
     def rules(self) -> List[Tuple[int, int]]:
         return settings.RATE_LIMITING_MIRROR_REALM_RULES
 
+    def __str__(self) -> str:
+        return self.realm.name or '<Missing name>'
 
 def rate_limit_mirror_by_realm(recipient_realm: Realm) -> None:
-    # Code based on the rate_limit_user function:
     entity = RateLimitedRealmMirror(recipient_realm)
-    ratelimited, time = is_ratelimited(entity)
+    ratelimited = rate_limit_entity(entity)[0]
 
     if ratelimited:
-        statsd.incr("ratelimiter.limited.%s.%s" % (type(recipient_realm),
-                                                   recipient_realm.id))
-        raise RateLimited()
-
-    try:
-        incr_ratelimit(entity)
-    except RateLimiterLockingException:
-        logger.warning("Email mirror rate limiter: Deadlock trying to "
-                       "incr_ratelimit for realm %s" % (recipient_realm.name,))
```

Comment:
Same with this change; it'd be nice to capture the `logging.warning` output and assert it in a test now that we're moving the abstraction layer.

(Also, as a general rule, we want `test-backend` to be output-clean, though fixing that is a somewhat larger project).

Reply:
Added a mock in the RateLimiterLockingException part of the test_mirror_worker_rate_limiting test